README.md

```diff
@@ -239,24 +239,463 @@ One can also do it manually:
+5. ✅ DO design for idempotency everywhere. ❌ DONT [return TMI](#dont-return-tmi) that the world should not be taking a dependency on. 
+6. ❌ DONT [use `Result`](#dont-result) or a per-Aggregate `type Error`. ✅ [DO use minimal result types per decision function](#do-simplest-result)
+7. ❌ DONT [expose your `Fold.State`](#dont-expose-state) outside your Aggregate.
+8. ❌ DONT be a slave to CQRS for all read paths. ✅ [DO `AllowStale`](#do-allowstale) 🤔 [CONSIDER `QueryCurrent`](#consider-querycurrent)
+9. ❌ [DONT be a slave to the Command pattern](#dont-commands) or Mediatr
+
+## High level
+
+### ❌ DONT have a global `Types.fs`
+
+F# really shines at succinctly laying out a high level design for a system; see [_Designing with types_ by Scott Wlaschin for many examples](https://fsharpforfunandprofit.com/series/designing-with-types/).
+
+For an event sourced system, if anything, this is even more true - it's not uncommon to be able to convey the key moving parts of a system in a manner that's legible for both technical and non-technical stakeholders.
+
+It's important not to take this too far though - ultimately as a system grows, the key constraint of the fact that Events ultimately need to be Grouped in Categories of Streams needs to become then organising function.
+
+<a name="global-dont-share-types"></a>
+### ❌ DONT share types across Aggregates / Categories
+
+There are sometimes legitimate when cases where two Aggregates have overlapping concerns. It can be very tempting to put the common types into a central place and Just Share the contracts. This should be avoided. Instead:
+
+```fs
+❌ DONT DO THIS
+
+// <Types.fs>
+module Domain.Types
+
+type EntityContext = { name: string; area: string }
+
+...
+
+// <Aggregate>.fs
+module Aggregate
+
+open Domain.Types
+
+module Events =
+
+    type Event =
+        | Created of {| creator: UserId; context: EntityContext |}
+        ...
+
+// <Aggregate2>.fs
+module Aggregate2
+
+module Events =
+
+    type Event =
+        | Copied of {| by: UserId; context: Types.EntityContext |}
+        ...
+```
+
+Instead:
+- have each `module <Aggregate>` have its own version of each type that will be used in an event _within its `module Events`_. (The `decide` function can map from an input type if desired, but the important thing is that the Aggregate will need to be able to roundtrip its types in perpetuity, and having to disentangle the overlap between more than on Aggregate is simply never a good tradeoff) 
+- [sharing id types is fine](#global-do-share-ids)
+
+<a name="global-do-share-ids"></a>
+### ✅ DO have global strongly typed ids
+
+While [sharing the actual types is a no-no](#global-dont-share-types), having common id types is perfectly reasonable
+
+```fsharp
+module Domain.Types
+
+type UserId = ...
+type TenantId = ...
+
+...
+
+module Domain.User
+
+module Events =
+
+    type Joined = { tenant: TenantId; authorizedBy: UserId }
+
+```
+
+- CONSIDER UMX for non-serialized ids
```

Comment:
what does it mean for an ID to be `non-serialised`?

Reply:
I don't round-trip the ids in the equinox (or dotnet-templates) samples to a store, so:
- serialization wrt FsCodec bodies doesn't matter
- xss and model binding matter more than they would as they are used in dtos